### ISSUE TITLE
Misleading note in the tutorial

### ISSUE THREAD

**Issue by Kinrany** [reactjs/reactjs.org#1013]
This note from the tutorial: [reactjs.org/tutorial/tutorial.html#functional-components
](https://reactjs.org/tutorial/tutorial.html#functional-components)

> **Note**
> When we modified the Square to be a functional component, we also changed `onClick={() => this.props.onClick()}` to a shorter `onClick={props.onClick}` (note the lack of parentheses on *both* sides). In a class, we used an arrow function to access the correct `this` value, but in a functional component we don’t need to worry about `this`.

The note seems to suggest that `onClick={this.props.onClick}` wouldn't work in a `class` component, but that's false.

**Comment by gaearon** [comment]
Yeah.. The thing is, `onClick={this.props.onClick}` would work but `onClick={this.handleClick}` wouldn't (unless you bind it first). The tutorial doesn't go into binding so I thought explaining the difference would be even more confusing.

"In a class, we used an arrow function to access the correct this value" refers to an earlier version of the code that called `this.setState`. But by now the reader has probably forgotten about it.

What do you think we should change this to? Maybe we can just leave it as `() => props.onClick()` to skip the whole explanation.

**Comment by Kinrany** [comment]
> What do you think we should change this to? Maybe we can just leave it as `() => props.onClick()` to skip the whole explanation.

Makes sense! This is the only place in the whole tutorial where a function is not being wrapped in `()=>{}`.
